core/domain/rights_manager.py

```diff
@@ -242,6 +245,17 @@ def is_exploration_cloned(exploration_id):
     return bool(exploration_rights.cloned_from)
 
 
+def _update_first_published_datetime(exploration_id, first_published):
+    exploration_rights = get_exploration_rights(exploration_id)
+    exploration_rights.first_published = first_published
+    _update_activity_summary(ACTIVITY_TYPE_EXPLORATION, exploration_rights)
+
+def update_first_published_datetime_if_not_set(exploration_id, first_published):
```

Comment:
I don't think you need this? In what cases will we ever be setting the datetime wrongly?